### PR TITLE
LPS-182606 | Test Automation | Master

### PR DESCRIPTION
**Pull request by davisaints**
**JIRA Ticket: [LPS-182606](https://issues.liferay.com/browse/LPS-182606)**

### CODE REVIEW

**Comment by davisaints** [comment]
ci:test:sf

**Review by Luizcarlosqueiroz — APPROVED** [review]
LGTM! :tada:

**Comment by timpak** [comment]
Taking a look

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectNotifications.testcase [hunk 1]
```diff
@@ -382,6 +382,97 @@ definition {
 		}
 	}
 
+	@description = "LPS-165849 - Verify if terms from a related custom object (child) with a system object (parent) are replaced correctly in an email notification"
```
Comment (timpak) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectNotifications.testcase:385]: `Verify terms from a 1:M related system object (parent) are replaced with the correct values in an email notification`

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectNotifications.testcase [hunk 2]
```diff
@@ -382,6 +382,97 @@ definition {
...
+	@priority = 5
+	test CanSendEmailNotificationWithTermsComingFromSystemObjectRelationship {
```
Comment (timpak) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectNotifications.testcase:387]: CanSendEmailWithTermsFromRelatedSystemObject

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectNotifications.testcase [hunk 3]
```diff
@@ -382,6 +382,97 @@ definition {
...
+		property custom.properties = "feature.flag.LPS-165849=true";
+		property portal.acceptance = "true";
+		property test.smtp.server.enabled = "true";
+
+		task ("Given an object with a field is created") {
```
Comment (timpak) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectNotifications.testcase:392]: `Given a custom object with a text field`

Be specific with what you're adding

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectNotifications.testcase [hunk 4]
```diff
@@ -382,6 +382,97 @@ definition {
...
+				fieldLabelName = "Field Text",
+				fieldName = "customField",
+				fieldType = "String",
+				isRequired = "false",
+				objectName = "CustomObject182606");
+
+			ObjectAdmin.publishObjectViaAPI(objectName = "CustomObject182606");
+		}
+
+		task ("And given the User object with a 1:M relationship to custom object") {
```
Comment (timpak) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectNotifications.testcase:409]: And given a 1:M relationship between User and the custom object

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectNotifications.testcase [hunk 5]
```diff
@@ -382,6 +382,97 @@ definition {
...
+			ObjectAdmin.addObjectRelationshipViaAPI(
+				objectName_1 = "User",
+				objectName_2 = "CustomObject182606",
+				relationshipLabel = "Relationship",
+				relationshipName = "relationship",
+				relationshipType = "oneToMany");
+		}
+
+		task ("And given a email notification template using terms from both object definitions") {
```
Comment (timpak) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectNotifications.testcase:418]: And given an email notification template using terms from both object definitions

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectNotifications.testcase [hunk 6]
```diff
@@ -382,6 +382,97 @@ definition {
...
+
+		task ("Given an object with a field is created") {
+			ObjectAdmin.addObjectViaAPI(
+				labelName = "Custom Object 182606",
+				objectName = "CustomObject182606",
+				pluralLabelName = "Custom Objects 182606");
+
+			ObjectAdmin.addObjectFieldViaAPI(
+				fieldBusinessType = "Text",
+				fieldLabelName = "Field Text",
```
Comment (timpak) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectNotifications.testcase:400]: `Text Field`

Adjective goes before noun

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectNotifications.testcase [hunk 7]
```diff
@@ -382,6 +382,97 @@ definition {
...
+				fromAddress = "[%CUSTOMOBJECT182606_AUTHOR_EMAIL_ADDRESS%]",
+				fromName = "[%RELATIONSHIP_USER_AUTHOR_FIRST_NAME%]",
+				name = "Notification Template",
+				notificationType = "Email",
+				subject = "[%RELATIONSHIP_USER_AUTHOR_FIRST_NAME%] [%RELATIONSHIP_USER_AUTHOR_LAST_NAME%]");
+
+			PortletEntry.save();
+		}
+
+		task ("And given an action in custom object to send an email notification after an entry is added") {
```
Comment (timpak) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectNotifications.testcase:436]: And given an action in the custom object to send an email notification after an entry is created

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectNotifications.testcase [hunk 8]
```diff
@@ -382,6 +382,97 @@ definition {
...
+			ObjectAdmin.openObjectAdmin();
+
+			ObjectPortlet.selectCustomObject(label = "Custom Object 182606");
+
+			ObjectAdmin.goToActionsTab();
+
+			ObjectAdmin.addObjectActionViaUI(
+				actionLabel = "Notification action when entry is added",
```
Comment (timpak) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectNotifications.testcase:444]: Send email after entry is created

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectNotifications.testcase [hunk 9]
```diff
@@ -382,6 +382,97 @@ definition {
...
+				active = "true",
+				notificationName = "Notification Template",
+				objectLabel = "Custom Object 182606",
+				thenAction = "Notification",
+				whenAction = "On After Add");
+		}
+
+		task ("When an entry is added") {
```
Comment (timpak) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectNotifications.testcase:452]: When an entry is created

**Review by timpak — CHANGES_REQUESTED** [review]
@davisaints please see my comments below

**Comment by davisaints** [comment]
ci:test:Sf

**Comment by davisaints** [comment]
ci:test:sf

**Comment by davisaints** [comment]
Merged on https://github.com/liferay-bpm-qa/liferay-portal/pull/762

**Pull request closed without merging**